### ISSUE TITLE
bug(streaming): `update` sanity check failed in hash join

### ISSUE THREAD

**Issue by BugenZhao** [risingwavelabs/risingwave#5891]
With in-memory state store, so it should not be related to Hummock.

```
thread 'risingwave-streaming-actor' panicked at 'inconsistent update!
table_id: 1147, vnode: 169, key: Row([None, None])
stored value: None
expected value: Row([None, Some(Int64(0))])
new value: Row([None, Some(Int64(1))])', /risingwave/src/storage/src/table/streaming_table/state_table.rs:707:13
stack backtrace:
2022-10-18T03:36:42.636195Z DEBUG risingwave_stream::task::stream_manager: drop actors actors=[125, 126, 127, 128, 137, 138, 139, 140]
   0: rust_begin_unwind
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/panicking.rs:142:14
   2: risingwave_storage::table::streaming_table::state_table::StateTable<S>::do_update_sanity_check::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:707:13
   3: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
   4: risingwave_storage::table::streaming_table::state_table::StateTable<S>::batch_write_rows::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:620:84
   5: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
   6: risingwave_storage::table::streaming_table::state_table::StateTable<S>::commit::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:570:57
   7: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
   8: risingwave_stream::executor::managed_state::join::JoinHashMap<K,S>::flush::{{closure}}
             at ./src/stream/src/executor/managed_state/join/mod.rs:459:39
   9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
  10: risingwave_stream::executor::hash_join::HashJoinExecutor<K,S,_>::flush_data::{{closure}}
             at ./src/stream/src/executor/hash_join.rs:721:36
  11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
  12: risingwave_stream::executor::hash_join::HashJoinExecutor<K,S,_>::into_stream::{{closure}}
             at ./src/stream/src/executor/hash_join.rs:678:52
```

https://buildkite.com/risingwavelabs/pull-request/builds/10391#0183e923-a39b-40d2-b917-05518e432f1a

I guess this is the degree table? cc @yuhao-su Could you please take a look?

**Comment by yuhao-su** [comment]
I'll take a look...

**Comment by chenzl25** [comment]
Closed by #6007.
